### ISSUE TITLE
Object Disposed Exception

### ISSUE THREAD

**Issue by andrewmd5** [deniszykov/WebSocketListener#9]
I'm running into this issue occasionally that doesn't seem to be catchable 

```
ObjectDisposedException: Cannot access a disposed object.
Object name: 'System.Net.Sockets.NetworkStream'.
  File "E:\Dev\WebSocketListener\vtortola.WebSockets\Extensibility\WebSocketSecureConnectionExtension.cs", line 47, col 13, in MoveNext
    Void MoveNext()
  Module "System.Runtime.CompilerServices.TaskAwaiter", line 40, col 0, in HandleNonSuccessAndDebuggerNotification
    Void HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
  Module "System.Runtime.ExceptionServices.ExceptionDispatchInfo", line 12, col 0, in Throw
    Void Throw()
  Module "System.Threading.Tasks.TaskFactory`1", line 25, col 0, in FromAsyncCoreLogic
    Void FromAsyncCoreLogic(System.IAsyncResult, System.Func`2[System.IAsyncResult,TResult], System.Action`1[System.IAsyncResult], System.Threading.Tasks.Task`1[TResult], Boolean)
  Module "System.Net.Security.SslState", line 116, col 0, in EndProcessAuthentication
    Void EndProcessAuthentication(System.IAsyncResult)
  Module "System.Net.Security.SslState", line 36, col 0, in InternalEndProcessAuthentication
    Void InternalEndProcessAuthentication(System.Net.LazyAsyncResult)
```

**Comment by deniszykov** [comment]
I will make this exception 'observed' in near update. So you wouldn't need to do anything.

**Comment by deniszykov** [comment]
fixed in 0ab7bd5973ab0951a0ba4df82a2da0f1765bdfa4. New version is in Nuget
